packages/quantic/force-app/main/default/lwc/quanticRefineToggle/quanticRefineToggle.js

```diff
@@ -0,0 +1,182 @@
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * Indicates whether the Quantic Sort component should be hidden.
+   * @api
+   * @type {boolean}
+   */
+  @api hideSort;
+  /**
+   * Indicates whether the refine modal will be opened in full screen.
+   * @api
+   * @type {boolean}
+   * @defaultValue `false`
+   */
+  @api fullScreen = false;
+
+  /** @type {QuerySummary} */
+  querySummary;
+  /** @type {BreadcrumbManager} */
+  breadcrumbManager;
+  /** @type {number} */
+  total;
+  /** @type {number} */
+  activeFiltersCount = 0;
+  /** @type {string} */
+  modalId = 'refineModal';
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {SearchEngine} engine
+   */
+  initialize = (engine) => {
+    this.querySummary = CoveoHeadless.buildQuerySummary(engine);
+    this.breadcrumbManager = CoveoHeadless.buildBreadcrumbManager(engine);
+
+    this.unsubscribeQuerySummary = this.querySummary.subscribe(() =>
+      this.updateTotalResults()
+    );
+    this.unsubscribeBreadcrumbManager = this.breadcrumbManager.subscribe(() =>
+      this.updateActiveFiltersCount()
+    );
+  };
+
+  disconnectedCallback() {
+    this.unsubscribeQuerySummary?.();
+    this.unsubscribeBreadcrumbManager?.();
+  }
+
+  /**
+   * Updates total number of results.
+   * @returns {void}
+   */
+  updateTotalResults() {
+    this.total = this.querySummary.state.total;
+  }
+
+  /**
+   * Updates the number of active filters.
+   * @returns {void}
+   */
+  updateActiveFiltersCount() {
+    this.activeFiltersCount = this.breadcrumbManager.state.hasBreadcrumbs
+      ? this.getFiltersCountFromAllBreadcrumbs()
+      : 0;
+  }
+
+  /**
+   * Returns the number of active filters from all the breadcrumbs.
+   * @returns {number}
+   */
+  getFiltersCountFromAllBreadcrumbs() {
+    const facetBreadcrumbKeys = [
+      'facetBreadcrumbs',
+      'numericFacetBreadcrumbs',
+      'categoryFacetBreadcrumbs',
+      'dateFacetBreadcrumbs',
+      'staticFilterBreadcrumbs',
+    ];
+    return facetBreadcrumbKeys.reduce(
+      (count, facetKey) =>
+        count +
+        this.getFiltersCountFromFacetBreadcrumb(
+          this.breadcrumbManager?.state?.[facetKey]
+        ),
+      0
+    );
+  }
+
+  /**
+   *  Returns the number of active filters from a specific facet breadcrumb.
+   * @param {Array<{values: Array}>} facetBreadcrumb
+   */
+  getFiltersCountFromFacetBreadcrumb(facetBreadcrumb) {
+    if (!facetBreadcrumb) {
+      return 0;
+    }
+    return facetBreadcrumb.reduce(
+      (count, facet) => count + facet.values.length,
+      0
+    );
+  }
+
+  /**
+   * Opens the refine modal.
+   * @returns {void}
+   */
+  openModal() {
+    /** @type {QuanticModalElement} */
+    const modal = this.template.querySelector(`[data-id=${this.modalId}]`);
```

Comment:
Might want to have a `this.getModal()` function?

Reply:
sure 👌